### DISCUSSION TITLE
TIKTOK:: Provider "tiktok" is missing both issuer and token endpoint config. At least one of them is required

### DISCUSSION THREAD

**Discussion by JohnRSim** [nextauthjs/next-auth#9800] [Help]
### Provider type

Tiktok

### Environment

System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Memory: 21.91 GB / 31.89 GB
  Binaries:
    Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.17.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 120.0.6099.225
    Edge: Chromium (120.0.2210.144)
    Internet Explorer: 11.0.19041.3636
  npmPackages:
    @auth/core: ^0.21.0 => 0.21.0
    @auth/mongodb-adapter: ^2.0.13 => 2.0.13
    @auth/sveltekit: ^0.7.0 => 0.7.0

### Reproduction URL

https://github.com/JohnRSim/authIssue

### Describe the issue

I'm getting the following auth error:
looking at the provider the issuer is not available but it does expose the token as a function - it looks valid and so am unsure why it is being flagged:

```
{
  id: 'tiktok',
  name: 'TikTok',
  type: 'oauth',
  authorization: {
    url: 'https://www.tiktok.com/v2/auth/authorize',
    params: {
      client_key: '---',
      scope: 'user.info.profile',
      response_type: 'code'
    }
  },
  token: { request: [AsyncFunction: request] },
  userinfo: {
    url: 'https://open.tiktokapis.com/v2/user/info/?fields=open_id,avatar_url,display_name,username',
    request: [AsyncFunction: request]
  },
  profile: [Function: profile],
  style: { logo: '/tiktok.svg', bg: '#000', text: '#fff' },
  options: {
    userinfo: 'https://open-api.tiktok.com/user/info/',
    clientId: '---',
    clientSecret: '---'
  }
}
```

[auth][error] InvalidEndpoints: Provider "tiktok" is missing both issuer and token endpoint config. At least one of them is required. .Read more at [https://errors.authjs.dev#invalidendpoints](https://errors.authjs.dev/#invalidendpoints)
at assertConfig (file:///C:/work/hivar.be/hivar/node_modules/@auth/core/lib/utils/assert.js:70:24)
at Module.Auth (file:///C:/work/hivar.be/hivar/node_modules/@auth/core/index.js:70:29)
at async getSession (C:/work/hivar.be/hivar/node_modules/@auth/sveltekit/index.js:215:22)
at async authorization (C:/work/hivar.be/hivar/src/hooks.server.js:48:18)
Error: There was a problem with the server configuration. Check the server logs for more information.
at getSession (C:\work\hivar.be\hivar\node_modules@auth\sveltekit\index.js:216:11)
at async authorization (C:\work\hivar.be\hivar\src\hooks.server.js:52:18)
at async Module.respond (C:\work\hivar.be\hivar\node_modules@sveltejs\kit\src\runtime\server\respond.js:296:20)
at async file:///C:/work/hivar.be/hivar/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:502:22

### How to reproduce

Setup Sveltekit add the TikTok provider to the server.hooks and run sveltekit
Error displayed in console

### Expected behavior

Error should not be displayed as I can see the token function is there and issuer is not required only token or issuer is required.

**Comment by balazsorban44** [comment]
Unfortunately I tried before and could not get a TikTok client set up, they never gave me access. So I can only guess. We are asserting the config, and looks like you are missing the `token.url`:

https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/utils/assert.ts/#L114

Since this is not a built-in provider, it's not a bug per se, so I'll move this to discussions. If you have the time, it would be awesome to have TikTok as a built-in provider!

**Comment by JohnRSim** [comment]
Hi @balazsorban44  - thanks for taking a look I'll have a look at the token.url and get it added and test again.

I found the provider here at auth.js when you say it is not built in - where is the official list
https://authjs.dev/reference/core/providers/tiktok

Many thanks

**Reply by balazsorban44** [reply]
Oh, so we do have it :sweat_smile:. My bad. I guess I was just remembering to not be able to test it due to not passing the app review. I have reapplied, hopefully it goes through this time!

![image](https://github.com/nextauthjs/next-auth/assets/18369201/567081b8-9190-4d9c-9484-13202a69b1ee)


Source code is at: https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/providers/tiktok.ts

The reason I assumed it was not built-in, because your config option did not seem to be wrapped with the official one. Once I have a working client, I could much more easily verify any issues. :crossed_fingers:

**Reply by balazsorban44** [reply]
Well, I tried... This is very annoying.
![image](https://github.com/nextauthjs/next-auth/assets/18369201/a697b3ec-66df-421b-9f68-e0bbc66ba844)

**Reply by balazsorban44** [reply]
Wrote them an email for clarification. I could not find this info anywhere.

**Reply by eidriahn** [reply]
so how the hell is one supposed to have a test env? XD

**Reply by JohnRSim** [reply]
@balazsorban44 if you want you can have mine (msged you on linkedin)
- I have it approved not using it at the moment?

**Reply by kilimchoi** [reply]
@JohnRSim would you be open to sharing the info? sent you a connection request on linkedin.

**Comment by JohnRSim** [comment]
Any progress here?

**Reply by mwawrusch** [reply]
I am running into the same issue.

**Comment by DewinU** [comment]
tiktok provider is broken rn for v5, see https://github.com/nextauthjs/next-auth/pull/11057,

**Reply by DewinU** [reply]
managed to get a workaraound, see #[11308](https://github.com/nextauthjs/next-auth/pull/11308)

**Reply by DewinU** [reply]
https://youtu.be/M-PCU-e8IGI?si=uunGbOxhNXIRlnXT

**Reply by DewinU** [reply]
#12168
